File: android/app/src/main/java/bh/pray/app/OnboardingCompletionStore.java
package bh.pray.app;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Versioned first-launch onboarding completion. Separate from presentation so it can be unit-tested.
 */
public final class OnboardingCompletionStore {
    public static final String PREFS_NAME = "pray_bh_onboarding";
    public static final String KEY = "onboarding.v2.complete";

    private final SharedPreferences prefs;

    public OnboardingCompletionStore(Context context) {
        this(context.getApplicationContext().getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE));
    }

    public OnboardingCompletionStore(SharedPreferences prefs) {
        this.prefs = prefs;
    }

    public boolean isComplete() {
        return prefs.getBoolean(KEY, false);
    }

    public void markComplete() {
        prefs.edit().putBoolean(KEY, true).apply();
    }
}

File: ios/PrayBHApp/OnboardingCompletionStore.swift
import Combine
import Foundation

/// Versioned first-launch onboarding completion. Separate from presentation so it can be unit-tested.
@MainActor
final class OnboardingCompletionStore: ObservableObject {
    static let key = "onboarding.v2.complete"

    private let defaults: UserDefaults
    @Published private(set) var isComplete: Bool

    init(defaults: UserDefaults = .standard) {
        self.defaults = defaults
        self.isComplete = defaults.bool(forKey: Self.key)
    }

    func markComplete() {
        defaults.set(true, forKey: Self.key)
        isComplete = true
    }
}
